### DISCUSSION TITLE
How to install it on Linux? go install github.com/xjasonlyu/tun2socks/v2@latest; tun2socks ---> bash: tun2socks: command not found

### DISCUSSION THREAD

**Discussion by slrslr** [xjasonlyu/tun2socks#295] [Q&A]
Hello,

at https://github.com/xjasonlyu/tun2socks/wiki/Install-from-Source is the installation command `go install github.com/xjasonlyu/tun2socks/v2@latest` which I have ran on Debian 11 (old-stable) Linux with result:

```
go: downloading github.com/xjasonlyu/tun2socks v1.18.3
go: downloading github.com/xjasonlyu/tun2socks/v2 v2.5.1
go: downloading github.com/docker/go-units v0.5.0
go: downloading go.uber.org/automaxprocs v1.5.2
go: downloading github.com/sirupsen/logrus v1.9.3
go: downloading go.uber.org/atomic v1.11.0
go: downloading gvisor.dev/gvisor v0.0.0-20230603040744-5c9219dedd33
go: downloading golang.org/x/sys v0.8.0
go: downloading golang.org/x/time v0.3.0
go: downloading github.com/Dreamacro/go-shadowsocks2 v0.1.8
go: downloading github.com/go-chi/chi/v5 v5.0.8
go: downloading github.com/go-chi/cors v1.2.1
go: downloading github.com/go-chi/render v1.0.2
go: downloading github.com/google/btree v1.1.2
go: downloading golang.org/x/crypto v0.9.0
go: downloading github.com/ajg/form v1.5.1
```

But it does not work further:
```
$ tun2socks --device tun://wg1 --proxy socks5://127.0.0.1:1080
bash: tun2socks: command not found
```

Which and whereis commands does not find tun2socks. I have wg1 network interface (Wireguard) which I want to use for one of my web browsers (which supports HTTPS and SOCKS5 proxy).

Please which commands to run in order to do it on my Debian 11 (old-stable) Linux?

**Comment by xjasonlyu** [comment]
You can download it from releases directly.

**Reply by slrslr** [reply]
This seems to be working to download/update latest version (in this case linux-amd64 one) from https://github.com/xjasonlyu/tun2socks/releases

`cd /usr/local/bin/ && wget -qO- https://github.com/xjasonlyu/tun2socks/releases/latest/download/tun2socks-linux-amd64.zip | sudo busybox unzip -o - && sudo chmod +x tun2socks-linux-amd64 && cd - && alias tun2socks="/usr/local/bin/tun2socks-linux-amd64"; sleep 3 && tun2socks --help`

Though utilizing my existing wg1 interface fails:
```
$ sudo /usr/local/bin/tun2socks* --device tun://wg1 --proxy socks5://127.0.0.1:1080
FATA[0000] [ENGINE] failed to start: create tun: invalid argument
```

[ip l;ip r output](https://bin.disroot.org/?4238f44ba173678d#F5Si4QCzNsis51a4S4mSj2Ui35e13dC39HQZLq8FJhKn)

When I try to set "-interface wg1":
```
$ sudo netstat -tulnp|grep 108;sudo /usr/local/bin/tun2socks* --device tun://tun5 -interface wg1 --proxy socks5://127.0.0.1:1085
INFO[0000] [DIALER] bind to interface: wg1              
INFO[0000] [STACK] tun://tun5 <-> socks5://127.0.0.1:1085
```

and in web browser set socks5 proxy 127.0.0.1:1085 , browser says "proxy server is refusing connections". As mentioned, I have wg1 network interface (Wireguard) which I want to use for one of my web browsers (which supports HTTPS and SOCKS5 proxy), please.

**Reply by slrslr** [reply]
```
$ ping -c 2 -I 10.2.0.2 1.1.1.1
PING 1.1.1.1 (1.1.1.1) from 10.2.0.2 : 56(84) bytes of data.
64 bytes from 1.1.1.1: icmp_seq=1 ttl=57 time=39.8 ms
64 bytes from 1.1.1.1: icmp_seq=2 ttl=57 time=32.6 ms

$ sudo /usr/local/bin/tun2socks* -interface wg1 --device tun://10.2.0.2 --proxy socks5://127.0.0.1:1081

INFO[0000] [DIALER] bind to interface: wg1              
INFO[0000] [STACK] tun://10.2.0.2 <-> socks5://127.0.0.1:1081
```
and setting browser's proxy to 127.0.0.1:1081 cause "The proxy server is refusing connections"

wg1 is my wireguard interface 10.2.0.2 that I want to use in one of my web browsers.

> 5: wg1: <POINTOPOINT,NOARP,UP,LOWER_UP> mtu 1340 qdisc noqueue state UNKNOWN group default qlen 1000
>     link/none 
>     inet 10.2.0.2/32 scope global wg1
>        valid_lft forever preferred_lft forever

```
$ ip r
default via 192.168.0.1 dev enx... proto static metric 100 
192.168.0.0/24 dev enx... proto kernel scope link src 192.168.0.5 metric 100
```

@xjasonlyu @cty123 @Amaindex please kindly let me know what to try

**Reply by slrslr** [reply]
@xjasonlyu @cty123 @Amaindex

**Reply by slrslr** [reply]
any solution?
